src/main/java/uk/ac/cam/cl/dtg/segue/api/managers/UserAccountManager.java

```diff
@@ -1999,4 +2007,80 @@ public Map<SchoolInfoStatus, Long> getSchoolInfoStats() throws SegueDatabaseExce
+    }
+    user.setTeacherPending(newFlagValue);
+    RegisteredUser updatedUser = database.createOrUpdateUser(user);
+    return dtoMapper.map(updatedUser, RegisteredUserDTO.class);
+  }
+
+  public void sendRoleChangeRequestEmail(final HttpServletRequest request, final RegisteredUserDTO user,
+                                         final Role requestedRole, final Map<String, String> requestDetails)
+      throws SegueDatabaseException, ContentManagerException, MissingRequiredFieldException {
+    String userSchool = getSchoolNameWithPostcode(user);
+    if (userSchool == null) {
+      throw new MissingRequiredFieldException(
+          String.format("School information could not be found for user with ID: %s", user.getId()));
+    }
+
+    String verificationDetails = requestDetails.get("verificationDetails");
+    if (verificationDetails == null || verificationDetails.isEmpty()) {
+      throw new MissingRequiredFieldException("No verification details provided");
+    }
+
+    String otherInformation = requestDetails.get("otherInformation");
+    String otherInformationLine;
+    if (otherInformation == null || otherInformation.isEmpty()) {
+      otherInformationLine = "";
+    } else {
+      otherInformationLine = String.format("Any other information: %s\n<br>\n<br>", otherInformation);
+    }
+
+    String roleName = requestedRole.toString();
+    String emailSubject = String.format("%s Account Request", roleName);
+    String emailMessage = String.format(
+        "Hello,\n<br>\n<br>"
+            + "Please could you convert my Isaac account into a teacher account.\n<br>\n<br>"
+            + "My school is: %s\n<br>"
+            + "A link to my school website with a staff list showing my name and email"
+            + " (or a phone number to contact the school) is: %s\n<br>\n<br>\n<br>"
+            + "%s"
+            + "Thanks, \n<br>\n<br>%s %s",
+        userSchool, verificationDetails, otherInformationLine, user.getGivenName(), user.getFamilyName());
+    Map<String, Object> emailValues = new ImmutableMap.Builder<String, Object>()
+        .put("contactGivenName", user.getGivenName())
+        .put("contactFamilyName", user.getFamilyName())
+        .put("contactUserId", user.getId())
+        .put("contactUserRole", user.getRole())
+        .put("contactEmail", user.getEmail())
+        .put("contactSubject", emailSubject)
+        .put("contactMessage", emailMessage)
+        .put("replyToName", String.format("%s %s", user.getGivenName(), user.getFamilyName()))
+        .build();
+    emailManager.sendContactUsFormEmail(properties.getProperty(Constants.MAIL_RECEIVERS), emailValues);
+
+    logManager.logEvent(user, request, SegueServerLogType.CONTACT_US_FORM_USED,
+        ImmutableMap.of("message", String.format("%s %s (%s) - %s", user.getGivenName(),
+            user.getFamilyName(), user.getEmail(), emailMessage)));
+  }
+
+  private String getSchoolNameWithPostcode(final RegisteredUserDTO user) {
```

Comment:
Is this new logic being tested somewhere? I cannot find the tests.

Reply:
I've written integration tests for the endpoint but I haven't gotten to unit tests for the new UserAccountManager functions yet. It's on the todo list, I just haven't had time this week to do so yet.

Reply:
Okay, got some units tests set up that should hopefully cover most of the new methods in UserAccountManager.
Jacoco doesn't seem to be recognising parameterised tests in the coverage report though so the numbers may be a little off... 🤔